ring/func/base.py

```diff
@@ -87,6 +97,10 @@ def coerce_function(t):
     if issubclass(t, (set, frozenset)):
         return _coerce_set
 
+    if sys.version_info >= (3, 7):
```

Comment:
Instead of version check, this is testable by existance of dataclasses.
Let's avoid direct version check as much as possible in code.

(setup.py and test codes are ok)

Reply:
`coerce_function` is a partial cache implementaiton of `coerce`. (Yes, I agree the naming is very ugly. it must be changed.) `dataclass` doesn't look fit for this case.
Moving this into `coerce` will be fine.